sdk/src/entrypoint.rs

```diff
@@ -42,7 +108,10 @@ macro_rules! entrypoint {
             unsafe {
```

Comment:
seems like the scope of this `unsafe` can be reduced.  For example the branches of the `match` are _safe_